src/main/java/org/embulk/filter/column/ColumnFilterPlugin.java

```diff
@@ -117,7 +120,7 @@ static Schema buildOutputSchema(PluginTask task, Schema inputSchema)
                 boolean matched = false;
                 for (ColumnConfig dropColumn : dropColumns) {
                     // skip json path notation to build outputSchema
-                    if (dropColumn.getName().startsWith("$.")) {
+                    if (PathCompiler.isDocContext(dropColumn.getName())) {
```

Comment:
Does it check only a string starts with dollar? What we want to do here is to filter paths which look like jsonpath.

So, It looks better to try `PathCompiler.compile` and put paths if no error occurs, rather than just checking starting dollar.